handlers/prometheus_handlers.go

```diff
@@ -1,282 +1,308 @@
-
-	if len(result) != len(boardFunc) {
-		http.Error(w, "unable to get static board", http.StatusInternalServerError)
-		return
-	}
-
-	err := json.NewEncoder(w).Encode(result)
-	if err != nil {
-		logrus.Errorf("error marshalling board: %v", err)
-		http.Error(w, "unable to marshal board instance", http.StatusInternalServerError)
-		return
-	}
-}
-
-// SaveSelectedPrometheusBoardsHandler persists selected board and panels
-func (h *Handler) SaveSelectedPrometheusBoardsHandler(w http.ResponseWriter, req *http.Request, _ *sessions.Session, prefObj *models.Preference, user *models.User, provider models.Provider) {
-	if req.Method != http.MethodPost {
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-
-	if prefObj.Prometheus == nil || prefObj.Prometheus.PrometheusURL == "" {
-		http.Error(w, "Prometheus URL is not configured", http.StatusBadRequest)
-		return
-	}
-
-	// if prefObj.Prometheus.SelectedPrometheusBoardsConfigs == nil {
-	// 	prefObj.Prometheus.SelectedPrometheusBoardsConfigs = []*models.GrafanaBoard{}
-	// }
-
-	defer func() {
-		_ = req.Body.Close()
-	}()
-
-	body, err := ioutil.ReadAll(req.Body)
-	if err != nil {
-		msg := "unable to read the request body"
-		logrus.Error(errors.Wrapf(err, msg))
-		http.Error(w, msg, http.StatusInternalServerError)
-		return
-	}
-	boards := []*models.SelectedGrafanaConfig{}
-	err = json.Unmarshal(body, &boards)
-	if err != nil {
-		msg := "unable to parse the request body"
-		logrus.Error(errors.Wrapf(err, msg))
-		http.Error(w, msg, http.StatusBadRequest)
-		return
-	}
-	if len(boards) > 0 {
-		prefObj.Prometheus.SelectedPrometheusBoardsConfigs = boards
-	} else {
-		prefObj.Prometheus.SelectedPrometheusBoardsConfigs = nil
-	}
-	err = provider.RecordPreferences(req, user.UserID, prefObj)
-	if err != nil {
-		logrus.Errorf("unable to save user config data: %v", err)
-		http.Error(w, "unable to save user config data", http.StatusInternalServerError)
-		return
-	}
-	_, _ = w.Write([]byte("{}"))
-}
+package handlers
+
+import (
+	"context"
+	"encoding/gob"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+
+	"github.com/gorilla/sessions"
+	"github.com/layer5io/meshery/helpers"
+	"github.com/layer5io/meshery/models"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func init() {
+	gob.Register(&models.PrometheusClient{})
+}
+
+// ScanPromGrafanaHandler - fetches  Promotheus and Grafana
+func (h *Handler) ScanPromGrafanaHandler(w http.ResponseWriter, req *http.Request, _ *sessions.Session, prefObj *models.Preference, user *models.User, provider models.Provider) {
+
+	if prefObj.K8SConfig == nil || !prefObj.K8SConfig.InClusterConfig && (prefObj.K8SConfig.Config == nil || len(prefObj.K8SConfig.Config) == 0) {
+		logrus.Error("No valid kubernetes config found.")
+		http.Error(w, `No valid kubernetes config found.`, http.StatusBadRequest)
+		return
+	}
+
+	installedMeshes, err := helpers.ScanPromGrafana(prefObj.K8SConfig.Config, prefObj.K8SConfig.ContextName)

```

Comment:
@vijcheru , Please change the variable name here, it should not be installed meshes, as it respresents grafana and prometheus instances. You could name it availablePromGrafama